File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/content/page/layout/LayoutComponentType.ts
module api.content.page.layout {

    import ComponentType = api.content.page.ComponentType;

    export class LayoutComponentType extends ComponentType {

        constructor() {
            super("layout");
        }

        newComponentBuilder(): LayoutComponentBuilder {
            return new LayoutComponentBuilder();
        }
    }

    new LayoutComponentType();

}

File: modules/core-api/src/main/java/com/enonic/xp/core/content/page/region/LayoutComponentType.java
package com.enonic.xp.core.content.page.region;

public final class LayoutComponentType
    extends ComponentType
{
    public final static LayoutComponentType INSTANCE = new LayoutComponentType();

    private LayoutComponentType()
    {
        super( "layout", LayoutComponent.class );
    }

}
